pkg/detectors/bulksms/bulksms.go

```diff
@@ -59,25 +59,30 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			}
 
 			if verify {
-				data := fmt.Sprintf("%s:%s", resIdMatch, resMatch)
-				sEnc := b64.StdEncoding.EncodeToString([]byte(data))
+				// data := fmt.Sprintf("%s:%s", resIdMatch, resMatch)
```

Comment:
Please remove these commented lines. Also I guess you missed the third point from my last review.

Reply:
Sure, I am unclear about the last point, are you talking about the tests? Can you please elaborate more?

Reply:
The secret in this context (Basic Auth token) is a combination of two components: an ID and a Key. Both the ID and Key are unique for each secret. Currently, we loop through each ID and attempt to verify it against all the available keys. 

For example, if we find 3 IDs and 3 Keys, we start by verifying the first ID with all the keys. If, on the first attempt, we successfully verify that the first ID and the first Key form a valid Basic Auth token, there is no need to check the remaining keys for that ID since it has already been verified. We can then move on to the next ID.

To further optimize this process, it would be beneficial to avoid comparing the subsequent IDs with any keys that have already been verified with previous IDs. This would reduce the number of unnecessary comparisons and improve efficiency by reducing the total number of combinations checked.

Reply:
The secret (Basic Auth token) is a combination of two components here, an ID and a Key. Both the ID and Key are unique for each secret as I observed after generating multiple tokens. Currently, we loop through each ID and attempt to verify it against all the available keys. [O(nxm)]

If we find 3 IDs and 3 Keys, we start by verifying the first ID with all the keys. If, on the first attempt, we successfully verify that the first ID and the first Key form a valid Basic Auth token, there is no need to check the remaining keys for that ID since it has already been verified. We can then move on to the next ID.

To further optimize this process, it would be more better to avoid comparing the remaining IDs with any keys that have already been verified with previous IDs. This would reduce the number of unnecessary comparisons and improve efficiency.

Reply:
Thanks, I have created a map of string -> boolean to mark the keys if a combination matches.